frontends/web/src/hooks/api.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2021 Shift Crypto AG
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useEffect, useState } from 'react'
+import { SubscriptionCallback } from '../api/subscribe';
+import { Unsubscribe } from '../utils/event';
+import { useMountedRef } from './utils';
+
+/**
+ * useSubscribe is a hook to subscribe to a subscription function.
+ * starts on first render, and returns undefined while there is no first response.
+ * re-renders on every update.
+ */
+export const useSubscribe = <T>(
+    subscription: ((callback: SubscriptionCallback<T>) => Unsubscribe)
+): (T | undefined) => {
+    const [respose, setResponse] = useState<T>();
+    const mounted = useMountedRef();
+    useEffect(
+        () => {
+            const unsubscribe = subscription((data) => {
+                if (mounted.current) {
+                    setResponse(data);
+                }
+            })
+            return unsubscribe;
+        }, // we pass no dependencies because it's only suscribed once
+        [] // eslint-disable-line react-hooks/exhaustive-deps
+    );
+    return respose;
+}
+
+
+/**
+ * useLoad is a hook to load a promise. 
+ * gets fired on first render, and returns undefined while loading.
+ */
+export const useLoad = <T>(
+    apiCall: () => Promise<T>
+): (T | undefined) => {
+    const [respose, setResponse] = useState<T>();
+    const mounted = useMountedRef();
+    useEffect(
+        () => {
+            apiCall().then((data) => {
+                if (mounted.current) {
+                    setResponse(data);
+                }
+            });
```

Comment:
Question: doesn't it need `return unsubscribe;` ?

Reply:
No, because it's not a subscription, it's a request only

Reply:
ah sorry, ofc, you are right :)